src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,954 @@
+        # [STEP_1]: Step #1.2 - Track OTA attributes: UpdateState and UpdateStateProgress
+        # [STEP_1]: Step #1.2.1 - UpdateState matcher: Track "Downloading"
+        # [STEP_1]: Step #1.2.2 - UpdateStateProgress matcher: Track non-null values "range 1–100" and final "None"
+        # ------------------------------------------------------------------------------------
+        logger.info(
+            f'{step_number}: Step #1.1 - Started subscription for UpdateState and UpdateStateProgress attributes '
+            'before AnnounceOTAProvider to avoid missing OTA events')
+
+        state_sequence = []
+        progress_values = []
+        downloading_seen = False
+        progress_seen = False
+
+        def matcher_combined(report):
+            """
+            Combined matcher for Step 1
+            - Validates UpdateState reaches Downloading
+            - UpdateStateProgress has any value 1–100
+            Follows the style of matcher_ota_updatestate.
+            """
+            nonlocal state_sequence, progress_values, downloading_seen, progress_seen
+            val = getattr(report.value, "value", report.value)
+
+            current_time = time.time()
+
+            # UpdateState
+            if report.attribute == Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState:
+                if val is not None and val == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading:
+                    if not downloading_seen:
+                        downloading_seen = True
+                        state_sequence.append(Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading)
+                        logger.info(f'{step_number}: State observed: {val} at {current_time}')
+
+            # UpdateStateProgress
+            elif report.attribute == Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateStateProgress:
+                if val is not None and isinstance(val, int) and 1 <= val <= 100:
+                    if not progress_seen:
+                        progress_seen = True
+                        progress_values.append(val)
+                        logger.info(f'{step_number}: Progress observed: {val} at {current_time}')
+
+            # End matcher once minimal conditions met
+            return downloading_seen and progress_seen
+
+        matcher_combined_obj = AttributeMatcher.from_callable(
+            description=f"{step_number} - Minimal Step 1 matcher: Downloading + progress 1-100",
+            matcher=matcher_combined
+        )
+
+        # ------------------------------------------------------------------------------------
+        # [STEP_1]: Step #1.3 - Start tasks to track OTA attributes:
+        # UpdateState and UpdateStateProgress (updateAvailable sequence) with validations
+        # ------------------------------------------------------------------------------------
+        subscription_attr.await_all_expected_report_matches([matcher_combined_obj], timeout_sec=60.0)
```

Comment:
```suggestion
        subscription_attr.await_all_expected_report_matches([matcher_combined_obj], timeout_sec=800.0)
```
i know this seems long, but the spec recommendation allows devices up to 600 seconds to send the query after getting an urgent update announcement. The extra time is for starting the download. I don't know how much fudge is actually reasonable here, but 200 extra seconds seems like enough for any device to at least start a download.

Reply:
`timeout_sec` has been updated as suggested.